zrml/orderbook/src/tests.rs

```diff
@@ -107,367 +541,506 @@ fn it_places_orders() {
-                base_asset_amount: 4300,
+                maker_asset,
+                // from 100 to 86 changed (partially filled) minus fees
+                maker_amount: filled_maker_amount,
+                taker_asset,
+                // from 500 to 430 changed (partially filled)
+                taker_amount: taker_amount - alice_portion,
             }
         );
 
-        let reserved_bob = Tokens::reserved_balance(outcome_asset, &BOB);
-        assert_eq!(reserved_bob, 860);
+        let reserved_bob = AssetManager::reserved_balance(maker_asset, &BOB);
+        assert_eq!(reserved_bob, filled_maker_amount);
 
         System::assert_last_event(
             Event::<Runtime>::OrderFilled {
                 order_id,
                 maker: BOB,
                 taker: ALICE,
-                filled: 700,
-                unfilled_outcome_asset_amount: 860,
-                unfilled_base_asset_amount: 4300,
+                filled_taker_amount: alice_portion,
+                unfilled_maker_amount: filled_maker_amount,
+                unfilled_taker_amount: taker_amount - alice_portion,
             }
             .into(),
         );
 
-        let alice_bal = <Balances as Currency<AccountIdTest>>::free_balance(&ALICE);
-        let alice_shares = Tokens::free_balance(outcome_asset, &ALICE);
-        assert_eq!(alice_bal, BASE - 700);
-        assert_eq!(alice_shares, 140);
+        let alice_taker_asset_free = AssetManager::free_balance(taker_asset, &ALICE);
+        let alice_maker_asset_free = AssetManager::free_balance(maker_asset, &ALICE);
+        assert_eq!(alice_taker_asset_free, INITIAL_BALANCE - alice_portion);
+        assert_eq!(
+            alice_maker_asset_free,
+            Perbill::from_rational(alice_portion, taker_amount).mul_floor(maker_amount)
+        );
+        assert_eq!(alice_maker_asset_free, 140_000_000_000);
 
-        let bob_bal = <Balances as Currency<AccountIdTest>>::free_balance(&BOB);
-        let bob_shares = Tokens::free_balance(outcome_asset, &BOB);
-        assert_eq!(bob_bal, BASE + 700);
-        // ask order was adjusted from 1000 to 860, and bob had 2000 shares at start
-        assert_eq!(bob_shares, 1000);
+        let bob_taker_asset_free = AssetManager::free_balance(taker_asset, &BOB);
+        let bob_maker_asset_free = AssetManager::free_balance(maker_asset, &BOB);
+        let filled_minus_fees =
+            alice_portion - ExternalFees::<Runtime, FeeAccount>::get_fee(market_id, alice_portion);
+        assert_eq!(bob_taker_asset_free, INITIAL_BALANCE + filled_minus_fees);
+        assert_eq!(bob_maker_asset_free, 0);
 
-        let reserved_bob = Tokens::reserved_balance(outcome_asset, &BOB);
-        assert_eq!(reserved_bob, 860);
+        let reserved_bob = AssetManager::reserved_balance(maker_asset, &BOB);
+        assert_eq!(reserved_bob, filled_maker_amount);
     });
 }
 
 #[test]
-fn it_removes_orders() {
+fn it_removes_order() {
```

Comment:
This test seems to be mixing multiple tests (success and failure). I think we should split this.